k8s/k8s.go

```diff
@@ -316,6 +445,29 @@ func overrideIPAMResult(ipAddrsNoIpam string, logger *log.Entry) (*types.Result,
 	return &result, nil
 }
 
+// parseIPAddrs is a utility function that parses string of IPs in json format that are
+// passed in as a string and returns a slice of string with IPs.
+// It also makes sure the slice isn't empty.
+func parseIPAddrs(annotation, ipAddrsStr string, logger *log.Entry) ([]string, error) {
```

Comment:
The point I'm trying to make is one about breaking the code into logical components.  As it is, the lines between this code and the rest are quite blurred because the code in this function is based on our knowledge that the calling code is always going to be from an annotation. 

What about if we want to call this code based on something from the CNI network config, which isn't an annotation?  What value would we pass for `annotation`?